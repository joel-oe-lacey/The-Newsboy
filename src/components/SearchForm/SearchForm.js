import React, { Component } from 'react';
import './SearchForm.scss';

class SearchForm extends Component {
    constructor() {
      super()
      this.state = {
        searchTerm: ''
      }
    }

    updateSearchTerm = event => {
        this.setState({searchTerm: event.target.value})
    }

    runSearch = event => {
        event.preventDefault();
        this.props.searchCards(this.state.searchTerm);
        this.setState({searchTerm: ''});
    }
    
    render() {
    return (
        <form className="searchArea">
            <input
                type="text"
                name="search"
                placeholder="Search for news articles here."
                value={this.state.searchTerm}
                onChange={this.updateSearchTerm}
            />
            <button onClick={this.runSearch}>Search</button>
        </form>
    )
    }
}

export default SearchForm;