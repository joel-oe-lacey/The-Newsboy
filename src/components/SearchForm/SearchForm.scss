.searchArea {
  grid-area: search;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;

  input {
    height: 100%;
    width: 40%;
    background: transparent;
    border: none;
    font-size: 1.5em;
    color: white;
  }

  button {
    height: 50%;
    width: 10%;
    background: #F26419;
    box-shadow: inset 0px 0px 0px white;
    border-radius: 5%;
    border: none;
    color: white;
    font-size: 1.5em;
    transition: all 0.4s ease 0s;

    &:hover {
      box-shadow: inset 180px 0px 0px #F26419;
      transition: all 0.4s ease 0s;
      cursor: pointer;
    }
  }
}