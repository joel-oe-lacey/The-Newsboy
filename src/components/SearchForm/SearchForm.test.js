import React from "react";
import { shallow } from "enzyme";
import SearchForm from "./SearchForm";

describe("SearchForm", () => {
  let wrapper;
  let mockSearchCards = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<SearchForm searchCards={mockSearchCards}/>);
  });

  it("renders a form for news searching", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should keep track of search terms as entered using updateSearchTerm", () => {
    const mockEvent = {
        target: {
            value: "testPhrase"
        }
    };

    wrapper.instance().updateSearchTerm(mockEvent);
    expect(wrapper.state("searchTerm")).toEqual("testPhrase");
  });

  it("should be able to kick off a search with the entered value using runSearch", () => {
    const mockEvent1 = {
        target: {
            value: "testPhrase"
        }
    };
    const mockEvent2 = { preventDefault: jest.fn() };

    wrapper.instance().updateSearchTerm(mockEvent1);
    wrapper.instance().runSearch(mockEvent2);

    expect(mockSearchCards).toHaveBeenCalledWith("testPhrase");
    expect(wrapper.state("searchTerm")).toEqual("");
  });
});
