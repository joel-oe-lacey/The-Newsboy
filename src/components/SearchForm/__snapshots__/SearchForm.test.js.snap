// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders a form for news searching 1`] = `
<form
  className="searchArea"
>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search for news articles here."
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
